File: 0179_Largest Number/LargestNumber.cs
public class Solution {
    public string LargestNumber(int[] nums) {
        Array.Sort(nums, (t1,t2)=>{
            var s1 = t1.ToString() + t2.ToString();
            var s2 = t2.ToString() + t1.ToString();
            return s2.CompareTo(s1);
        });

        var sb = new StringBuilder();
        foreach(var v in nums){
            sb.Append(v);
        }

        return sb[0] == '0' ? "0" : sb.ToString();
    }
}

File: 0179_Largest Number/LargestNumber.java
class Solution {
    public String largestNumber(int[] nums) {
    	nums = Arrays.stream(nums).boxed().sorted((a, b) -> {
    		String s1 = a + "" + b;
            String s2 = b + "" + a;
            return s2.compareTo(s1);
    	}). // sort descending
        mapToInt(i -> i).
        toArray();
    	
    	StringBuilder sb = new StringBuilder();
        for(int v : nums) {
            if(v == 0 && sb.length() == 0) continue;
            sb.append(v);
        }
        return sb.length() == 0 ? "0" : sb.toString();
    }
}